lib/adapters/MochaAdapter.js

```diff
@@ -6,10 +6,14 @@ export default class MochaAdapter extends EventEmitter {
     super()
 
     this.mocha = mocha
+    this.origReporter = mocha._reporter
 
     mocha.reporter((runner) => {
       this.runner = runner
 
+      let constructor = this.origReporter.bind(this.mocha, this.runner)
```

Comment:
Can you we `new this.origReporter.bind(this.mocha, this.runner)()` here? Along with suppressing whatever eslint rule will fail it.